parser/parser_test.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2015 The Serulian Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package parser
+
+import (
+	"container/list"
+	"fmt"
+	"io/ioutil"
+	"sort"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type testNode struct {
+	nodeType   NodeType
+	properties map[string]string
+	children   map[string]*list.List
+}
+
+type parserTest struct {
+	name     string
+	filename string
+}
+
+func (pt *parserTest) input() string {
+	b, err := ioutil.ReadFile(fmt.Sprintf("tests/%s.seru", pt.filename))
+	if err != nil {
+		panic(err)
+	}
+
+	return string(b)
+}
+
+func (pt *parserTest) tree() string {
+	b, err := ioutil.ReadFile(fmt.Sprintf("tests/%s.tree", pt.filename))
+	if err != nil {
+		panic(err)
+	}
+
+	return string(b)
+}
+
+func createAstNode(source InputSource, kind NodeType) AstNode {
+	return &testNode{
+		nodeType:   kind,
+		properties: make(map[string]string),
+		children:   make(map[string]*list.List),
+	}
+}
+
+func (tn *testNode) Connect(predicate string, other AstNode) AstNode {
+	if tn.children[predicate] == nil {
+		tn.children[predicate] = list.New()
+	}
+
+	tn.children[predicate].PushBack(other)
+	return tn
+}
+
+func (tn *testNode) Decorate(property string, value string) AstNode {
+	tn.properties[property] = value
+	return tn
+}
+
+var parserTests = []parserTest{
+	// Simple tests.
+	{"basic import test", "basic_import"},
+	{"from import test", "from_import"},
+	{"string import test", "string_import"},
+	{"multiple import test", "multiple_import"},
```

Comment:
Adding an error field and failure cases will make the test coverage even better!